src/components/directory-content.js

```diff
@@ -622,7 +633,11 @@ const DirectoryContent = () => {
     function toggleSelectAll() {
         if (selectedUuids.size === 0) {
             setSelectedUuids(
-                new Set(currentChildren.map((c) => c.elementUuid))
+                new Set(
+                    currentChildren
+                        .map((c) => c.elementUuid)
```

Comment:
Are those flipped ? I would have put the filtering first and then the mapping for I don't trust an uuid has a uploading property.